### DISCUSSION TITLE
LuckPerms Context

### DISCUSSION THREAD

**Discussion by xtomyserrax** [RoinujNosde/SimpleClans#200] [Ideas]
Im considering adding this on my own plugin so maybe it might be useful for someone using SC (not for me but idk)

LuckPerms added this context feature which is amazing! For example, you could create a context like if a player is in a clan or not.

So in LP, some permissions or ranks could be assigned if simpleclans:isInClan is true

[LP Context Wiki](https://github.com/lucko/LuckPerms/wiki/Context)

[EssentialsX release notes adding support for this](https://www.spigotmc.org/resources/essentialsx.9089/update?update=347136)

**Comment by xtomyserrax** [comment]
I basicly did add this and was quite useful. Some ideas could be (names could ofc be changed)
- isInClan
- isClanLeader (and those other ranks like untrusted, etc)
- isBannedFromClans
- isInPermanentClan ¿?
- isInClanWar
- isInVerifiedClan
- hasPlayerFriendlyFireEnable ¿?
- hasClanFriendlyFireEnable ¿?
- isInClanChat
- isInAlliesChat

And who knows what else

**Reply by RoinujNosde** [reply]
Hm, friendly fire too
